src/core/net/ip6.cpp

```diff
@@ -678,6 +725,15 @@ ThreadError Ip6::HandleDatagram(Message &message, Netif *netif, int8_t interface
     // process IPv6 Payload
     if (receive)
     {
+        if (nextHeader == kProtoIp6)
+        {
+            // Remove encapsulating header.
+            message.RemoveHeader(message.GetOffset());
```

Comment:
I'm not sure we want to blindly remove an encapsulating IPv6 header in case there is legitimate IPv6-in-IPv6 traffic flowing over the Thread network.  How above a rule where we do not pass up messages destined to the `ALL_MPL_FORWARDERS` address?  Passing up the inner packet should just work on the next iteration through `HandleDatagram()`.